src/main/java/com/demo/folder/repository/TraineeRepository.java

```diff
@@ -0,0 +1,61 @@
+package com.demo.folder.repository;
+
+import com.demo.folder.entity.Trainee;
+import org.hibernate.Session;
+import org.hibernate.SessionFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Repository;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.List;
+
+@Repository
+public class TraineeRepository {
+
+  @Autowired
+  private SessionFactory sessionFactory;
+
+  public TraineeRepository() {}
+
+  private Session getCurrentSession() {
+    return sessionFactory.getCurrentSession();
+  }
+
+  @Transactional
```

Comment:
While it's technically possible to use @ Transactional on a repository class, it's generally considered better practice to use it on a service layer.